R/R/consortia_internal_tracking.R

```diff
@@ -0,0 +1,219 @@
+# Synapse data tracking - track "working" data available to internal consortia. 
+# Note modified, new and deleted files.
+# Author: Kelsey Montgomery
+# Date: 2019.08.01
+#'
+#' Get Synapse view - fileview or table
+#'
+#' @param fileview_id A synId c().
+#' @export
+#' @return A tibble.
+#' @examples
+#' \dontrun{
+#' fv <- get_view("syn18691012")
+#' table <- get_view("syn20555115")
+#' }
+#' 
+get_view <- function(fileview_id) {
+  fileview <- synapser::synTableQuery(paste0("SELECT * FROM ", fileview_id))
+  readr::read_csv(fileview$filepath, col_types = readr::cols(.default = "c"))
+}
+#' Check whether updates need to be made
+#'
+#' @param fv A tibble. A fileview imported with [get_view()].
+#' @param table A tibble. A table imported with [get_view()].
+#' @param fileview_id A synId c().
+#' @export
+#' @return If no changes are to be made the string "No changes!" is returned. If 
+#' there are changes to be made to the existing table, a Synapse Table object is 
+#' returned with metadata to describe the upload.
+#' @examples
+#' \dontrun{
+#' update_table(fv = get_view("synId"), 
+#' table = get_view("synId"), 
+#' fileview_id = "syn20555115")
+#' }
+#'
+update_table <- function(fv, table, fileview_id) {
+  if (check_etag(fv, table) == c("No changes to existing annotations")
+      && check_version(fv, table) == c("No changes to existing files")
+      && setequal(fv$id, table$id[!(table$notes %in% c("File removed"))])) {
+    return("No changes!")
+  } else {
+    mod_table(fv, table, fileview_id)
+  }
+}
+#' Bind new files to the existing table, notes deleted files and file/annotation 
+#' modifications or checks only for file/annotation modifications. Deleted files
+#' are a special case. In order to not overwrite existing 'modifiedOn' notation,
+#' the object note_del_files stores a vector of synIds that corresponds to files 
+#' deleted prior. These entries will not be modified when mod_table() is
+#' executed.
+#'
+#' @param fv A tibble. A fileview imported with [get_view()].
+#' @param table A tibble. A table imported with [get_view()].
+#' @param fileview_id A synId c().
+#' @export
+#' @return A tibble with notation added to the notes column. ROW_ID and ROW_VERSION 
+#' remain from existing table to enable storage to Synapse.
+#' @examples
+#' \dontrun{
+#' mod_table(fv = get_view("synId"), table = get_view("synId"))
```

Comment:
This example is a little misleading because it doesn't include the `fileview_id` parameter, which is necessary for the function to work. Could you add that?

Reply:
The updated .Rd files are included in commit 32efd90